safe-api/src/type_rep.js

```diff
@@ -2,4 +2,7 @@
 export interface TypeRep<T> {}
 
 class StringRep implements TypeRep<string> {}
+class MaybeStringRep implements TypeRep<?string> {}
+
 export const string = new StringRep();
+export const maybeString = new MaybeStringRep();
```

Comment:
should `maybe` take a typerep and return a typerep, unioned with null and undefined? instead of encoding a `maybeFoo` for every type `Foo`.

Reply:
Yeah. Unless someone else is about to grab that I'd like that to be in the next PR. This is mostly to add a bunch of tests.